latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+        except FileNotFoundError:
+            return False
+        return not node_data.removed
+
+    @property
+    def name(self) -> str:
+        return get_node_data(self.path).data[self.path].name
+
+    @property
+    def type(self) -> LDataNodeType:
+        return get_node_data(self.path).data[self.path].type
+
+    def is_dir(self) -> bool:
+        return self.type in {
+            LDataNodeType.dir,
+            LDataNodeType.account_root,
+            LDataNodeType.mount,
+        }
+
+    @property
+    def size(self) -> float:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.size
+
+    @property
+    def content_type(self) -> str:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.content_type
+
+    def iterdir(self) -> Generator[Path, None, None]:
+        if not self.is_dir():
+            raise ValueError(f"Not a directory: {self.path}")
+        data = execute(
+            gql.gql("""
+            query LDataChildren($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise ValueError(f"No directory found at path: {self.path}")
+
+        for node in data["finalLinkTarget"]["childLdataTreeEdges"]["nodes"]:
+            yield urljoins(self.path, node["child"]["name"])
+
+    def rmr(self) -> None:
+        execute(
+            gql.gql("""
+            mutation LDataRmr($nodeId: BigInt!) {
+                ldataRmr(input: { argNodeId: $nodeId }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id},
+        )
+
+    def copy(self, dst: Union["LPath", str]) -> None:
+        remote_copy(self.path, str(dst))
+
+    def upload(self, src: Path, progress=Progress.tasks, verbose=False) -> None:
+        upload(src, self.path, progress, verbose)
+
+    def download(
+        self, dst: Optional[Path] = None, progress=Progress.tasks, verbose=False
+    ) -> Path:
+        if dst is None:
+            dir = Path(".") / "downloads" / str(uuid4())
+            dir.mkdir(parents=True, exist_ok=True)
+            dst = dir / self.name
+
+        download(self.path, dst, progress, verbose, confirm_overwrite=False)
+        return dst
+
+    @property
+    def perms(self) -> LDataPerms:
+        return get_node_perms(self.node_id)
+
+    def share_with(self, email: str, perm_level: PermLevel) -> None:
```

Comment:
would just remove this too unless we strictly need it

```suggestion
    def share_with(self, *, email: str, perm_level: PermLevel) -> None:
```

would also check email correctness with a sanity-check regex